src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1065,17 +1065,19 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
 
                                 // initializers can't produce state machines
                                 Debug.Assert((object)initializerStateMachineTypeOpt == null);
-
-                                Debug.Assert(processedInitializers.LoweredInitializers.Kind == BoundKind.StatementList);
                                 Debug.Assert(!hasErrors);
-                                hasErrors = processedInitializers.LoweredInitializers.HasAnyErrors || diagsForCurrentMethod.HasAnyErrors();
+                                hasErrors = lowered.HasAnyErrors || diagsForCurrentMethod.HasAnyErrors();
                                 SetGlobalErrorIfTrue(hasErrors);
 
                                 if (hasErrors)
                                 {
                                     _diagnostics.AddRange(diagsForCurrentMethod);
                                     return;
                                 }
+
+                                // Only do the cast if we haven't returned with some error diagnostics.
```

Comment:
> if lowered isn't a BoundStatement?

I don't think this can happen because LowerBodyOrInitializer returns BoundStatement, signature enforces this. 

However, it looks like LowerBodyOrInitializer doesn't always return BoundStatementList when it is given BoundStatementList as an input.

We have at least a couple of options:
- BoundStatementList is a BoundStatement, so we can change LowerBodyOrInitializer's type to be BoundStatement. This might not be desirable, depending on how LowerBodyOrInitializer is used elsewhere.
- We can always create a BoundStatementList with a single statement inside, the lowered, if lowered isn't already a BoundStatementList.